client/state-db/src/noncanonical.rs

```diff
@@ -350,6 +350,16 @@ impl<BlockHash: Hash, Key: Hash> NonCanonicalOverlay<BlockHash, Key> {
 		self.last_canonicalized.as_ref().map(|&(_, n)| n)
 	}
 
+	pub fn sync(&mut self) {
+		let mut pinned = std::mem::take(&mut self.pinned_canonincalized);
+		for hash in pinned.iter() {
+			self.unpin(hash)
+		}
+		pinned.clear();
```

Comment:
```suggestion
		self.pinned_canonincalized.drain(..).for_each(|h| self.unpin(h));
```

Reply:
This won't work because `drain` borrows `self` and `unpin` requires `&mut self`

Reply:
Good point, but the call to `clear` is also not really required.